astropy/modeling/mappings.py

```diff
@@ -55,17 +59,20 @@ def inverse(self):
                 "Mappings such as {0} that drop one or more of their inputs "
                 "are not invertible at this time.".format(self.mapping))
 
-        return self.__class__(mapping)
+        inv = self.__class__(mapping)
+        inv._inputs = self._outputs
+        inv._outputs = self._inputs
+        return inv
 
 
 class Identity(Mapping):
-    def __init__(self, n_inputs):
-        mapping = tuple(range(n_inputs))
+    def __init__(self, n_outputs):
+        mapping = tuple(range(n_outputs))
```

Comment:
I think they have a slightly different meaning, which makes it confusing. For example, 
`m = Mapping((1,), n_inputs=2)` means I will pass 2 inputs (but only the second one will be returned).
`ident = Identity(1)` does not really tell me how many inputs I will pass. It only means that the first one will be returned. `ident(2,3,4)` returns 3. `m.n_inputs` returns 2, while `ident.n_inputs` returns 1. So its meaning is really `n_outputs`.